article.md

```diff
@@ -0,0 +1,612 @@
+    hasComment: Sequelize.HasManyHasAssociationMixin<CommentInstance, CommentInstance["id"]>;
+    hasComments: Sequelize.HasManyHasAssociationsMixin<CommentInstance, CommentInstance["id"]>;
+    countComments: Sequelize.HasManyCountAssociationsMixin;
+
+    getAuthor: Sequelize.BelongsToGetAssociationMixin<UserInstance>;
+    setAuthor: Sequelize.BelongsToSetAssociationMixin<UserInstance, UserInstance["id"]>;
+    createAuthor: Sequelize.BelongsToCreateAssociationMixin<UserAttributes>;
+};
+```
+
+I'll let you stare at those for a second.
+
+## What about attributes?
+
+We also want to update our `ModelAttributes` interfaces so that 1) we can access associations on a model instance when we use eager loading, and 2) we can specify associations when we create a model. This is what we have to add.
+
+```typescript
+export interface CommentAttributes {
+  id?: number;
+  name: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+
+  // ** this is new **
+
+  // We allow both either PostInstance or a Post's primary key,
+  // so we can specifiy either when we create a model.
+  // `posts?` is optional because we don't want to force
+  // specifying associations when we create a model. We also
+  // want to be able to query for Comment's without also having
+  // to load its posts.
+  post?: PostInstance | PostInstance['id'];
+
+  // Similarly, we define the field `author?`. An `author` is an
+  // alias for the `User` model, so we define that `author?` can
+  // either be a `UserInstance` or a `UserInstance['id']`.
+  author?: UserInstance | UserInstance['id'];
+  
+  // `upvoters` is a BelongsToMany association, so we define that
+  // a comment can have an array of User's, under the field `upvoters`.
+  upvoters?: UserInstance[] | UserInstance['id'][];
+};
+```
+
+```typescript
+export interface UserAttributes {
+  id?: number;
+  name: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+
+  // ** this is new **
+  comments?: CommentInstance[] | CommentInstance['id'][];
+  posts?: PostInstance[] | PostInstance['id'][];
+  upvotedComments?: CommentInstance[] | CommentInstance['id'][];
+};
+```
+
+```typescript
+export interface PostAttributes {
+  id?: number;
+  name: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+
+  // ** this is new **
+  comments?: CommentInstance[] | CommentInstance['id'][];
+  author: UserInstance | UserInstance['id'];
+};
+```
+
+## Finally, the good stuff!
+
+Okay, we've just slaved away writing these big interfaces (or, I did, but you have a nice automated tool to do it for you). What was the point? Well, just look at these.
+
+![todo]
+
+The types Sequelize provide are well detailed, upto the most intricate settings in options objects.
+
+![todo]
+
+Type inference even works beautifully with enum types.
+
+![todo]
+
+> TODO: a nice little conclusion about how it's worth investing time into boilerplate for long-term codebase scalability, free documentation, and type-safety.
```

Comment:
glorious! yeah - I think I would partly add this to the beginning as a sort of "this takes a while, but is worthwhile for larger, more complex projects; if you're doing something simple, chances are this is overkill".